src/DynamoDbQueryBuilder.php

```diff
@@ -315,15 +325,23 @@ protected function conditionsContainKey()
         return $conditionValue;
     }
 
+    /**
+     * Check if conditions "where" contain index key
+     * For composite index key, it will return false if the conditions don't have all composite key.
+     *
+     * @return array|bool false or array of keys
+     */
     protected function conditionsContainIndexKey()
     {
         if (empty($this->where)) {
             return false;
         }
 
         foreach ($this->model->getDynamoDbIndexKeys() as $key => $name) {
-            if (isset($this->where[$key])) {
-                return $key;
+            $keys          = $this->explodedIndexKeys($key);
```

Comment:
@baopham Because, for composite index key, it will be declare like this:
```php
    protected $dynamoDbIndexKeys = [
        'index_key' => 'simple_index_name',
        'index_hash_key,index_range_key' => 'composite_index_name',
    ];
```
It also works with simple index key, of course.